hw/block/nvme.c

```diff
@@ -49,6 +49,44 @@
             " in %s: " fmt "\n", __func__, ## __VA_ARGS__); \
     } while (0)
 
+
+static void set_uncorrectable_blk(NvmeCtrl *n, NvmeNamespace *ns, NvmeCmd *cmd,
+    NvmeRequest *req)
+{
+    uint32_t nlb  = le32_to_cpu(cmd->nlb) + 1;
+    uint64_t slba = le64_to_cpu(cmd->slba);
+    if (unlikely((slba + nlb) > ns->id_ns.nsze)) {
+    block_acct_invalid(blk_get_stats(n->conf.blk), acct);
+    trace_nvme_err_invalid_lba_range(slba, nlb, ns->id_ns.nsze);
+    return NVME_LBA_RANGE | NVME_DNR;
+    }
+
+    for(int i=0; i<=nlba; i++){
+        ns->uncorrectable[slba+i]=1
```

Comment:
@yuval-lb you should use bitmap api, this loop is wrong cause only single bit allocated per lba,e.g

void bitmap_set(unsigned long *map, long i, long len);
void bitmap_set_atomic(unsigned long *map, long i, long len);
void bitmap_clear(unsigned long *map, long start, long nr);
bool bitmap_test_and_clear_atomic(unsigned long *map, long start, long nr);
void bitmap_copy_and_clear_atomic(unsigned long *dst, unsigned long *src,
                                  long nr);

Reply:
+1